incountry/validation/validate_model.py

```diff
@@ -1,83 +1,110 @@
-from typing import Dict, Union
+from typing import Callable, Dict, List, Union
+from functools import wraps, reduce
+from inspect import signature
 
-import wrapt
 from pydantic import BaseModel, ValidationError
 
 
-from .utils import function_args_to_kwargs, get_formatted_validation_error, path_to_obj
+from .utils import function_args_to_kwargs, get_formatted_validation_error
 from ..exceptions import (
     InputValidationException,
     StorageException,
-    StorageClientException,
     StorageServerException,
     StorageCountryNotSupportedException,
 )
 
 
-def get_res_model(model, class_instance):
-    res_model = model
-    if isinstance(model, dict) and isinstance(class_instance, object):
-        condition_value = path_to_obj(class_instance, model["condition"])
-        for option in model["values"]:
-            if condition_value == option["value"]:
-                res_model = option["model"]
-
-    return res_model
-
-
-def get_validated_data(function, model, exclude_to_dict, validation_exception=InputValidationException, **kwargs):
+def get_validated_data(function, model, validation_exception=InputValidationException, **kwargs):
     try:
-        return model.validate(kwargs).dict(exclude=exclude_to_dict, exclude_unset=True)
+        return model.validate(kwargs).dict(exclude_unset=True)
     except ValidationError as e:
         errors_report = get_formatted_validation_error(e)
         error_text = "Validation failed during {}():{}".format(function.__qualname__, errors_report)
         raise validation_exception(error_text) from None
 
 
-def validate_model(model: Union[BaseModel, Dict], exclude_to_dict={}, validation_exception=InputValidationException):
-    @wrapt.decorator
-    def decorator(function, instance, args, kwargs):
-        new_kwargs = dict(kwargs)
-        res_model = get_res_model(model, instance)
+def format_extra_fields_error(extra_fields):
+    return reduce(lambda agg, field_name: f"{agg} \n  {field_name} - extra fields not permitted", extra_fields, "")
```

Comment:
Pythonic way to do that (also it will probably reduce number of allocations):
```suggestion
    return " \n  ".join(
        f"{field_name} - extra fields not permitted"
        for field_name in extra_fields
    )
```